src/RequestExecutor/Request.spec.ts

```diff
@@ -0,0 +1,103 @@
+import 'reflect-metadata';
+import type { Request } from './Request';
+
+type RequestType = 'http' | 'ws';
+
+describe('Request', () => {
+  let readFileMock!: jest.Mock;
+
+  const requests: Record<RequestType, Request> = {
```

Comment:
`Request` is not aware that we have these implementations, they also seem to do nothing for the tests here. Please remove them and don't forget to adjust test cases a little bit :)